pkg/api/interceptor.go

```diff
@@ -163,28 +164,27 @@ func (wa *WalletAuthorizer) applyLimits(ctx context.Context, req interface{}, wa
 	}
 
 	// with no wallet apply regular limits
-	var priority bool
+	var isPriority bool
 	if len(wallet) > 0 && wa.AllowLists {
-		priority = wa.AllowLister.IsAllowListed(wallet.String())
+		isPriority = wa.AllowLister.IsAllowListed(wallet.String())
 	}
-
-	// default request cost is 1,
-	// note that nil req (subscription stream) has cost 1 as well
 	cost := 1
+	limitType := ratelimiter.DEFAULT
 	switch req := req.(type) {
 	case *messagev1.PublishRequest:
 		cost = len(req.Envelopes)
+		limitType = ratelimiter.PUBLISH
 	case *messagev1.BatchQueryRequest:
 		cost = len(req.Requests)
 	}
 	// need to separate the IP buckets between priority and regular wallets
 	var bucket string
-	if priority {
-		bucket = "P" + ip
+	if isPriority {
+		bucket = "P" + ip + string(limitType)
```

Comment:
I hope this isn't too subtle, but this is how we get different buckets for different limit types. I can see an argument that this should happen inside the limiter, but then the simple "give me the bucket name" parameter becomes something else ("bucket prefix"?). So that doesn't feel squeeky clean either.